HW02_A_Graph-Fails.Rmd

```diff
@@ -1,87 +1,97 @@
+
+It is also best coding practice to only call the packages you use, so if you use a package but end up tossing the code you use for it, then make sure to remove loading it in the first place. For example, I could use `library("tidyverse")` but since this script will only be using ggplot2, I only load ggplot2.
+
+```{r load libraries}
+library("ggplot2")
+library("magrittr") #so I can do some piping
+```
+
+
+### Graph Fail 1
+
+What error is being thrown? How do you correct it? (hint, the error message tells you)
+
+```{r}
+data(mpg) #this is a dataset from the ggplot2 package
+
+mpg %>% 
+  ggplot(mapping = aes(x = cty, y = hwy, color = "blue")) + 
+  geom_point()
+#changed city -> cty (corrected the spelling) and exchanged the %>% for the + sign
+```
+
+### Graph Fail 2
+Why aren't the points blue? It is making me blue that the points in the graph aren't blue :`(
+```{r}
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy),  color = "blue") 
+#needed to remove the argument for color outside of the aes parentheses
+
+```
+
+
+
+### Graph Fail 3
+Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
+
+Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+```{r}
+mpg %>% 
+ggplot() + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), size = 2) + 
+#alpha controls the transparency of the points, changing it to 2 does nothing since alpha operates on a % scale (it goes from 0 - 1)
+#changed the alpha argument to a size argument to make the points bigger
+  theme(legend.direction = "horizontal", 
+        legend.position = c(.6, .85), 
+        legend.title =   element_blank())
+#needed to change the numbers for the legend position to a 'relative position' (between 0 - 1), set the legend title to blank to remove it and put all of the legend code into one line to reduce redundancy
+#to remove the legend completely, you would set legend.position = "none"
+  
+```
+
+### Graph Fail 4
+I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
+
+```{r}
+mpg %>% 
+ggplot(mapping = aes(x = displ, y = hwy)) + 
+#needed to remove the color argument from the above line of code so the smooths aren't applied to each grouping individually
+  geom_point(aes(color = drv)) + 
+#by setting the color argument here instead, geom_smooth only acts on the data groupings as defined in the original ggplot line of code - so it fits one line because the data is one group
```

Comment:
I really like the versatility of ggplot2. When I removed the color argument to get the single smooth line, for a second I thought I lose the colors forever, but it is as simple as to specify them again in the geom_point() function.